File: backend/dataset/reverse/reverse_11.py
class Solution:
    def reverse(self, x: int) -> int:
        if(x<0):
            s=-int((str(-1*x)[::-1]))
            return s if(s>-2**31) else 0
        s=int((str(x)[::-1]))
        return s if (s<2**31-1) else 0

File: backend/CDataset/reverse/reverse_11.cpp
class Solution {
public:
    int reverse(int x) {
        int tut = x;
        if(x >= INT_MAX || x <= INT_MIN){
            return 0;
        }
		//makes positive
        if(x<0){
            tut = -tut;
        }
        string s = to_string(tut);
        ::reverse(s.begin(),s.end());
		//add '-' later
        if(x<0){
            s.insert(s.begin(),'-');
        }
        try{
            x = stoi(s);
        }
        catch(const std::out_of_range& e){
            return 0;
        }
        return x;
    }
};
